File: rb/features.rb
# NewsPublic SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module NewsPublicFeatures
  def self.make_feature(name)
    case name
    when "base"
      NewsPublicBaseFeature.new
    when "test"
      NewsPublicTestFeature.new
    else
      NewsPublicBaseFeature.new
    end
  end
end

File: py/features.py
# NewsPublic SDK feature factory

from feature.base_feature import NewsPublicBaseFeature
from feature.test_feature import NewsPublicTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NewsPublicBaseFeature(),
        "test": lambda: NewsPublicTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
